test/Release/MysqlReleaseGatewayTest.php

```diff
@@ -0,0 +1,161 @@
+<?php
+/**
+ * @author: Ryan Wood <ryanw@clearbooks.co.uk>
+ * @created: 10/08/15
+ */
+
+namespace Clearbooks\LabsMysql\Release;
+
+
+use Clearbooks\Labs\Release\Release;
+use Doctrine\DBAL\Configuration;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\DriverManager;
+
+class MysqlReleaseGatewayTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var MysqlReleaseGateway
+     */
+    private $gateway;
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $connectionParams = array(
```

Comment:
All right, fair enough, it's ok to do it later.

Hm I'm not too experienced myself either, but I don't think that setting up the connection should be the responsibility of the table. On first thought I think I would rather do it with some kind of a connection provider class which could setup the connection object and give it to you. What do you think @craigjbass?

And yes, the connection details should be in a config file somewhere, most probably not commited to git. Maybe a sample file only, and the real connection data should only exist on the server somewhere and put to the right place in deploy time or something similar.